src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java

```diff
@@ -0,0 +1,57 @@
+package ca.corefacility.bioinformatics.irida.ria.web.services;
+
+import java.util.Locale;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.stereotype.Component;
+
+import ca.corefacility.bioinformatics.irida.exceptions.IridaWorkflowNotFoundException;
+import ca.corefacility.bioinformatics.irida.model.workflow.IridaWorkflow;
+import ca.corefacility.bioinformatics.irida.model.workflow.description.IridaWorkflowDescription;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.dto.launch.UIPipelineDetailsResponse;
+import ca.corefacility.bioinformatics.irida.service.workflow.IridaWorkflowsService;
+
+/**
+ * UI Service for all things related to workflow pipelines.
+ */
+@Component
+public class UIPipelineService {
+    private final IridaWorkflowsService workflowsService;
+    private final MessageSource messageSource;
+
+    @Autowired
+    public UIPipelineService(IridaWorkflowsService workflowsService, MessageSource messageSource) {
+        this.workflowsService = workflowsService;
+        this.messageSource = messageSource;
+    }
+
+    /**
+     * Get the information about a specific workflow pipeline
+     *
+     * @param id     for a {@link IridaWorkflow}
+     * @param locale current users {@link Locale}
+     * @return Details contained within a {@link UIPipelineDetailsResponse}
+     * @throws IridaWorkflowNotFoundException exception thrown if the workflow cannot be found.
+     */
+    public UIPipelineDetailsResponse getPipelineDetails(UUID id, Locale locale) throws IridaWorkflowNotFoundException {
+        IridaWorkflow workflow = workflowsService.getIridaWorkflow(id);
+        IridaWorkflowDescription description = workflow.getWorkflowDescription();
+        UIPipelineDetailsResponse detailsResponse = new UIPipelineDetailsResponse();
+
+        /*
+        Prefix for getting messages from IRIDA message properties file
+         */
+        String prefix = "workflow." + description.getAnalysisType().getType() + ".";
+
+        /*
+        Set up basic information for the pipeline being launch.
```

Comment:
Change `launch` to `launched`

Reply:
Fixed in e8fd739f2696876b708027f27a68e7651df9e7ac